### ISSUE TITLE
'extends' is not compiling when used inside declaration file and importHelpers option is switched on

### ISSUE THREAD

**Issue by Nipheris** [Microsoft/TypeScript#12724]
**TypeScript Version:**  2.1.4 / 2.2.0-dev.20161207

**Code**

```ts
interface Animal {
	type: string;
}

export declare class Pet implements Animal {
	public type: string;
	public name: string;
	constructor();
}

export declare class Dog extends Pet {
	public breed: string;
	constructor();
}
```

**Expected behavior:**
Compiles successfully.

**Actual behavior:**
Compilation fails with the following error:
```
error TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.
```

The same source code compiles fine on TypeScript 2.1.1. Installed `tslib` package has version 1.2.0.

**Comment by mhegazy** [comment]
are you using `--importHelpers`?

**Comment by mhegazy** [comment]
never mind, saw the title :)

**Comment by mhegazy** [comment]
The error should not be shown in the .d.ts file, this a bug.

but if you have `tslib` installed you should not see the error anyways, is this correct?

**Comment by timocov** [comment]
@mhegazy yeah, this is correct

**Comment by PavelPZ** [comment]
I cannot run **importHelpers** with installed ```tslib```.

**TypeScript Version**: 2.1.4 vith VS.NET

**Code**
export const o = { a: 1, name: "o" };
export const copy = { ...o };

**tsconfig.json**
```
{
  "compilerOptions": {
    "target": "es5",
    "module": "system",
    "importHelpers": true,
    "lib": [
      "dom",
      "es6"
    ],
  }
}
```

**installation""
I use both JSPM and NPM
```
jspm install tslib
npm install tslib --save
```

**package.json**
```
{
  "jspm": {
    "dependencies": {
      "tslib": "npm:tslib@^1.2.0",
    }
  "dependencies": {
    "tslib": "^1.2.0"
  }
}
```

**Comment by mhegazy** [comment]
@PavelPZ, the compiler uses the same module resolution strategy to resolve `tslib` as it does with any other module. so you will need to either  1. add `"moduleResolution": "node",` or 2. add a path mapping for it: `"paths": { "tslib" : ["./node_modules/tslib/tslib.d.ts"]  }`

**Comment by PavelPZ** [comment]
@mhegazy, thanks.

I am using **JSPM**, so I modified my tsconfig.json:
```
{
  "compilerOptions": {"baseUrl": ".",
     "paths": {
        "tslib": [ "jspm_packages/npm/tslib@1.2.0/tslib.d.ts" ]
     }, 
```
and compiler is now OK.

But there is a problem with runtime (```Uncaught (in promise) Error: (SystemJS) tslib.__assign is not a function``` in Chrome).
```tslib.js``` is correctly loaded, but its factory is never called. 

**jspm.config.js**
```
SystemJS.config({
  paths: {
    "tslib": "jspm_packages/npm/tslib/tslib",
```

**Comment by mhegazy** [comment]
Fix for the original issue should be in tonight `typescript@next`.

**Comment by mhegazy** [comment]
@PavelPZ i have not had time to look into the jspm issue yet. have you made progress past this issue, or are you still blocked on it?

**Comment by PavelPZ** [comment]
@mhegazy, no progress so far. I have no idea how to solve my observation ```tslib.js is correctly loaded, but its factory is never called. So e.g. tslib.__assign is undefined```.

**Comment by mhegazy** [comment]
Filed https://github.com/Microsoft/TypeScript/issues/12887 to track it.

**Comment by prencher** [comment]
We are seeing this with 2.1.4 using Webpack 2 with awesome-typescript-loader. It only happens when using the importHelpers. The above workaround using paths does not work:
```
ERROR in [at-loader] node_modules\@types\react\index.d.ts:186:31
    This syntax requires an imported helper but module 'tslib' cannot be found.
```

**Comment by mhegazy** [comment]
This should be fixed in latest, can you give `typescript@next` a try?

**Comment by prencher** [comment]
@mhegazy Just saw that in a related ticket. It is indeed fixed in `next`, so we've just reverted the change for now. Thanks!

**Comment by schmuli** [comment]
@mhegazy I just tested this issue, using the latest version `2.2.0-dev.20161227`, and I'm still seeing this error.

`package.json`:

```json
{
  "name": "tslib-test",
  "version": "1.0.0",
  "dependencies": {
    "@angular/core": "^2.4.1",
    "tslib": "^1.4.0",
    "typescript": "2.2.0-dev.20161217",
    "rxjs": "^5.0.2",
    "zone.js": "0.7.4"
  }
}
```

`tsconfig.json`:

```json
{
    "compilerOptions": {
        "target": "es5",
        "module": "es6",
        "moduleResolution": "node",
        "noEmitHelpers": true,
        "importHelpers": true,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "lib": ["es6", "dom"]
    }
}
```

`main.ts`:
```javascript
import { NgModule } from '@angular/core';

@NgModule({

})
export class MyModule {

}
```

Error in Console:
```
$ node_modules/typescript/bin/tsc -p .
node_modules/@angular/core/src/metadata/di.d.ts(178,48): error TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.
```

**Comment by mhegazy** [comment]
thanks @schmuli for the repro, fix up in https://github.com/Microsoft/TypeScript/pull/13189/files

**Comment by Delagen** [comment]
When it will be published? in 2.2.1 is the same

**Comment by mhegazy** [comment]
@Delagen we believe the issue should be fixed in TS 2.2.1, if you are still seeing it please file a new issue and give us some information to reproduce the issue locally.

**Comment by code-chris** [comment]
Created new issue for 2.2.1: #14468
